core/src/main/java/io/opencensus/stats/ViewData.java

```diff
@@ -16,114 +16,134 @@
 import com.google.auto.value.AutoValue;
 import io.opencensus.common.Function;
 import io.opencensus.common.Timestamp;
-import io.opencensus.stats.View.DistributionView;
-import io.opencensus.stats.View.IntervalView;
-import java.util.ArrayList;
 import java.util.Collections;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 import javax.annotation.concurrent.Immutable;
 
 /**
  * The aggregated data for a particular {@link View}.
  */
 @Immutable
+@AutoValue
 public abstract class ViewData {
+
+  // Prevents this class from being subclassed anywhere else.
+  ViewData() {
+  }
+
   /**
    * The {@link View} associated with this {@link ViewData}.
    */
   public abstract View getView();
 
   /**
-   * Applies the given match function to the underlying data type.
+   * The {@link AggregationData}s grouped by {@link TagValue}s, associated with this
+   * {@link ViewData}.
+   *
+   * <p>Note: The returned map is unmodifiable, attempts to update it will throw an
+   * UnsupportedOperationException.
    */
-  public abstract <T> T match(
-      Function<? super DistributionViewData, T> p0,
-      Function<? super IntervalViewData, T> p1);
+  public abstract Map<List<TagValue>, List<AggregationData>> getAggregationMap();
 
-  // Prevents this class from being subclassed anywhere else.
-  private ViewData() {
+  /**
+   * Returns the {@link WindowData} associated with this {@link ViewData}.
+   *
+   * @return the {@code WindowData}.
+   */
+  public abstract WindowData getWindowData();
+
+  /**
+   * Constructs a new {@link ViewData}.
+   */
+  public static ViewData create(View view, Map<List<TagValue>, List<AggregationData>> map,
+      WindowData windowData) {
```

Comment:
Should this method do any validation, such as checking that the Window and WindowData types match?